cmd/main_test.go

```diff
@@ -301,3 +307,86 @@ var _ = DescribeTable("TestRunAccessCheck", func(user string, namespace string,
 		"patch",
 		false),
 )
+
+var _ = DescribeTable("TestGetNamespacesWithAccess", func(allNamespaces []k8sapi.Namespace,
+	expectedNs []k8sapi.Namespace, actualNs []k8sapi.Namespace, err error) {
+	e := echo.New()
+	cfg, _ := config.GetConfig()
+	clientset, _ := kubernetes.NewForConfig(cfg)
+	req := httptest.NewRequest(http.MethodGet, "/", nil)
+	rec := httptest.NewRecorder()
+	c := e.NewContext(req, rec)
+	c.Request().Header.Set("X-Email", "user1@konflux.dev")
+	authCl := clientset.AuthorizationV1()
+	
+	JustBeforeEach(func() {
+		actualNs, err = getNamespacesWithAccess(e, c, authCl, allNamespaces)
+	})
+
+	Context("When all the namespaces have all the necessary permissions like create, list, watch and delete", func() {
+		BeforeEach(func() {
+			namespaceNames := []string{"test-tenant", "test-tenant-2"}
+			for _, name := range namespaceNames {
+				ns, err := createNamespace(k8sClient, name)
+				Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Error while creating the namespace %s: %v", name, err))
+				allNamespaces = append(allNamespaces, ns)
+				createRole(k8sClient, name, "namespace-access", []string{"create", "list", "watch", "delete"})
+				createRoleBinding(k8sClient, "namespace-access-user-binding", name, "user1@konflux.dev", "namespace-access")
+			}			
+			expectedNs = allNamespaces
+		})
+		It("returns all namespaces in the list", func() {
+			Expect(actualNs).To(Equal(expectedNs))
+			Expect(err).NotTo(HaveOccurred(), "Unexpected error testing GetNamespacesWithAccess")
+		})
+		AfterEach(func() {
+			allNamespaces = []k8sapi.Namespace{}
```

Comment:
Those are not the variables you need to clean, 
what we need to clean are the actual resources.

for example:
in the test you ran the following- 
```
createNamespace(k8sClient, name)
createRole(k8sClient, name, "namespace-access", []string{"create", "list", "watch", "delete"})
createRoleBinding(k8sClient, "namespace-access-user-binding", name, "user1@konflux.dev", "namespace-access")
```

those resources may stay available on the envtest cluster while the other tests are running, as we use the same env. we should remove all resources we created.